main.rb

```diff
@@ -0,0 +1,147 @@
+module Enumerable
+  # 1.my_each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+    for item in self
+      yield(item)
+    end
+  end
+
+  # 2.my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+    index = 0
+    my_each do |item|
+      yield(item, index)
+      index += 1
+    end
+  end
+
+  # 3.my_select
+  def my_select
+    return to_enum(:my_select) unless block_given?
+    select_arr_result = []
+    each do |item|
+      select_arr_result << item if yield(item)
+    end
+    select_arr_result
+  end
+
+  # 4.my_all?
+  def my_all?(_parameter = nil)
+    length_array = size
+    true_counter1 = 0
+    true_counter = 0
+    # Check if there is block given
+    if block_given?
+      my_each do |x|
+        true_counter += 1 if yield(x)
+      end
+      true_counter == length_array
+    else
+      my_each do |element|
+        true_counter1 += 1 if !element.nil? == true
+      end
+      true_counter1 == length_array
+    end
+  end
```

Comment:
## my_all?:

- [ ] when no block or argument is given returns false when one of the collection members is false or nil:

```ruby

      false_array = [1, false, 'hi', []]
      false_array.my_all? === false_array.all? => false

```

- [ ] when a class is passed as an argument returns false if any of the collection is not a member of such class:

```ruby

      array = ["word", 23, 3, 98]
      array.my_all? === array.all? => false

```

- [ ] when a Regex is passed as an argument returns false if any of the collection does not match the Regex:

```ruby

      array = ["word1", "word2", "word3"]
      words.my_all?(/o/) === words.all?(/o/) => false

```

- [ ] when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern:

```ruby

      array = [21, 22, 54, 34]
      array.my_all?(3) === array.all?(3) => false

```